jsx/Form.js

```diff
@@ -145,6 +145,75 @@ FormElement.defaultProps = {
   },
 };
 
+/**
+ * FieldsetElement Component.
+ * React wrapper for <fieldset> element that is nested inside <FormElement></FormElement>,
+ * and accepts children react components. A fieldset groups related elements in a form.
+ *
+ * The form elements can be passed by:
+ * 1. Form components nested directly inside <FieldsetElement></FieldsetElement>
+ *
+ */
+class FieldsetElement extends Component {
+  constructor(props) {
+    super(props);
+    this.getFormElements = this.getFormElements.bind(this);
+  }
+
+  getFormElements() {
+    const formElementsHTML = [];
+    const columns = this.props.columns;
+    const maxColumnSize = 12;
+    const colSize = Math.floor(maxColumnSize / columns);
+    const colClass = 'col-xs-12 col-sm-' + colSize + ' col-md-' + colSize;
+
+    // Render elements from React
+    React.Children.forEach(this.props.children, function(child, key) {
+      // If child is plain HTML, insert it as full size.
+      // Useful for inserting <hr> to split form sections
+      let elementClass = 'col-xs-12 col-sm-12 col-md-12';
+
+      // If child is form element use appropriate size
+      if (React.isValidElement(child) && typeof child.type === 'function') {
+        elementClass = colClass;
+      }
+      formElementsHTML.push(
+        <div key={'el_child_' + key} className={elementClass}>{child}</div>
+      );
+    });
+    return formElementsHTML;
+  }
+
+  render() {
+    // Generate form elements
+    let formElements = this.getFormElements();
+
+    return (
+      <div style={{padding: '0px', margin: '0px'}}>
```

Comment:
I don't see the point of this div. The fieldset is already a block level element.

(I'm not entirely sure that there's much benefit in having this component at all instead of just having the CSS and putting the <fieldset> tag directly into the jsx components where you want them..)

Reply:
oops, you're right. i dont know why i thought that div was needed as a block level element.

fieldsets group related elements in a form, and so i see the benefit of having the ability to render multiple fieldsets in one FormElement

Reply:
Right, I agree, I just don't see what benefit `<FieldsetElement>xxx</FieldsetElement>` adds over `<fieldset>xxx</fieldset>` in that regard.